workflow/gccontroller/gc_controller.go

```diff
@@ -82,6 +97,75 @@ func (c *Controller) runWorker() {
 	}
 }
 
+type count struct {
+	errored int
+	failed  int
+}
+
+// retentionGC queues workflows for deletion based upon the retention policy.
+func (c *Controller) retentionGC() {
```

Comment:
This is very similar to `enforceHistoryLimit` in Cron. Can you replicate the implementation here? That implementation is based on how K8s itself performs, so we would rather go that route.

https://github.com/argoproj/argo-workflows/blob/34e5b54779b25416d7dbd41d78e0effa523c1a21/workflow/cron/operator.go#L298-L334

Reply:
Do you refer to move the logic from the Cron controller here or the opposite @simster7 ?

Reply:
Based upon the suggestion from @alexec this will not be relevant so will resolve it.